### PR TITLE
Check kind state when adding super class. #502

### LINKED ISSUE #502: Throw when superclass is used on an interface.
This is currently silently ignored.

### PR DESCRIPTION
**Pull request by sormuras**
`superclass` needs `Kind.CLASS`

Closes #502.

### CODE REVIEW

File: src/test/java/com/squareup/javapoet/TypeSpecTest.java [hunk 1]
```diff
@@ -2014,6 +2014,25 @@ private String toString(TypeSpec typeSpec) {
     }
   }
 
+  @Test public void superClassOnlyValidForClasses() {
+    TypeSpec.classBuilder("C").superclass(Object.class);
```
Comment (JakeWharton) [inline src/test/java/com/squareup/javapoet/TypeSpecTest.java]: This line isn't needed. It's already tested elsewhere to be working.

**Comment by JakeWharton** [comment]
lgtm

File: src/test/java/com/squareup/javapoet/TypeSpecTest.java (hunk 1 above)
Comment (sormuras) [inline src/test/java/com/squareup/javapoet/TypeSpecTest.java]: Line removed.

**Comment by JakeWharton** [comment]
Thanks!

**Pull request merged**